api/backend/infrastructure/repositories/ingestion.py

```
@@ -0,0 +1,425 @@
+        if sanitized_name != city_name:
+            raise ValueError("Invalid city_name: contains unsafe characters")
+
+        base = Path(file_path).resolve()
+        auth_path = Path(auth_file_path).resolve()
+
+        # Verify paths are within expected directory
+        from shared.paths import cities_pbf_dir
+
+        expected_base = cities_pbf_dir().resolve()
+        if not str(base).startswith(str(expected_base)):
+            raise ValueError(f"PBF file path outside expected directory: {base}")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)